src/browser/core_state.ts

```diff
@@ -211,7 +211,12 @@ export function getUuidBySourceUrl(sourceUrl: string): string|boolean {
 
 export function getConfigUrlByUuid(uuid: string): string|boolean {
     const app = getAppAncestor(uuid);
-    return app && app._configUrl;
+    if  (app && app._configUrl) {
+        return app._configUrl;
+    } else {
+        const externalApp = getExternalAncestor(uuid);
+        return externalApp && externalApp.configUrl;
```

Comment:
Allowed return values are `string|boolean`. This would have to be `return !!externalApp && externalApp.configUrl;`

Reply:
this function should not be returning `string|boolean` and falsey values should not be coerced to boolean. instead they should be coerced to `undefined` via:

```
// if getExternalAncestor is written correctly, this coercion isn't necessary.
const externalApp = getExternalAncestor(uuid) || {};

// returns a string value or undefined
return externalApp && externalApp.configUrl; 
```

Reply:
I like this better 👍

Reply:
... typescript reflects what we've had, and string|boolean is what we've had. Unless we start refactoring throughout, just safer to stick to what's there now?